gphys/gdalet.F

```diff
@@ -0,0 +1,161 @@
+	parameter (rhom=770.0)
+C
+	vdmform = 0.0
+	if (x.ge.rmas) return
+        if (x.lt.2.0*0.511) return
+
+        f = sqrt(1.-4.*(0.511/x)**2) * (1.+2.*(0.511/x)**2)
+        if (isw.eq.1) then       ! pseudo-scalar meson -> gamma e+ e-
+	   dalit = ((1.-(x/rmas)**2)**3/(x/rmas))*1./(1.-(x/rhom)**2)**2
+	   vdmform = 8.2e-4*dalit*f
+        else if (isw.eq.2) then  ! vector meson -> pi0 e+ e-
+           delta = (135.0/rmas)**2
+           bracket = (1.+(x/rmas)**2/(1.-delta))**2
+     +                - 4.*(x/rmas)**2/(1.-delta)**2
+           if(bracket.le.0.0) return
+           ff2 = 0.64**4/((0.65**2-(x/1000.)**2)**2 + (0.042*0.65)**2)
+           dalit = (bracket**1.5)/(x/rmas)*ff2
+           vdmform = 8.2e-4*dalit*f
+        end if
+C
+        return
+	end
+c------------------------------------------------------------------------------
+	real function qedform(x,rmas,isw) ! QED formfactor
+	real*4 x, rmas, dalit		  ! masses in MeV
+        real*4 f, delta
+C
+	qedform = 0.0
+	if (x.ge.rmas) return
+        if (x.lt.2.0*0.511) return
+
+        f = sqrt(1.-4.*(0.511/x)**2) * (1.+2.*(0.511/x)**2)
+        if (isw.eq.1) then       ! pseudo-scalar meson
+	   dalit = ((1.-(x/rmas)**2)**3/(x/rmas))
+	   qedform = 8.2e-4*dalit*f
+        else if (isw.eq.2) then  ! vector meson
+           delta = (135.0/rmas)**2
+           bracket = (1.+(x/rmas)**2/(1.-delta))**2
+     +                - 4.*(x/rmas)**2/(1.-delta)**2
+           if(bracket.le.0.0) return
+           dalit = (bracket**1.5)/(x/rmas)
+           qedform = 8.2e-4*dalit*f
+        end if
+C
+        return
+	end
+c------------------------------------------------------------------------------
+
+c------------------------------------------------------------------------------
+c Generation of 2 particle mass according to the VDM form factor
+c for Dalitz decay of (id) 1=pi0 2=eta  to (e+e-) gamma and 3=omega to pi0 e+e-
+c------------------------------------------------------------------------------
+
+      real function dalitz_evgen(id)
+      real xmin(3), xmax(3), wmax(3), mesonmass(3), rndm(2)
+      integer isw(3)
+      data mesonmass/134.9766,547.75,782.59/
+      data isw/1,1,2/
+      data xstep/0.1/
+      logical maxfound(3)
+      data maxfound/3*.FALSE./
+      save maxfound, wmax
+
+      if (.NOT. maxfound(id)) then
+         wmax(id) = 0.0
+         xmas    = 0.0
+ 1       xmas    = xmas + xstep
+         ff = vdmform(xmas,mesonmass(id),isw(id))
+         if (ff .gt. wmax(id)) wmax(id) = ff
+         if (xmas .lt. mesonmass(id)) goto 1
+         maxfound(id) = .TRUE.
+      end if
+
+ 2    call grndm(rndm,2)
+      rnd1 = rndm(1)*mesonmass(id)
+      rnd2 = rndm(2)*wmax(id)
+      ff = vdmform(rnd1,mesonmass(id),isw(id))
+      if (ff .lt. rnd2) goto 2
+      
+      dalitz_evgen = rnd1/1000.
+
+      return
+      end
+
+c------------------------------------------------------------------------------
+
+      SUBROUTINE GDALET(XM0,XM1,XM2,XM3,PCM,IDH)
+C.
```

Comment:
Where does this function come from? It is good to add a comment to the description below.